### ISSUE TITLE
How to declare additional attribute for a partial?

### ISSUE THREAD

**Issue by ceremcem** [ractivejs/ractive#3419]
Is there a known way to declare additional attributes for a partial? For example: 

```html
{{#partial mybutton}}
  <mycomponent foo="{{value}}" bar="{{size}}"  color="green" hello="there" /> 
  <!-- mycomponent also supports width= attribute -->
{{/partial}}

{{#with myvar1}}
  {{> mybutton}}
{{/with}}

{{#with myvar2}}
  {{> mybutton width="300"}} <- this syntax would be pretty handy
{{/with}}
```

I simply want to  use `mycomponent` with some default configuration and then I want to extend it with a new attribute only for the next instance.

**Comment by marcalexiei** [comment]
You can explicit provide context to the partial instead of wrapping it in a `with`:

```html
{{>mybutton myvar1}}
```

and after you can use object spread to pass additional attributes:

```html
{{>mybutton { ...myvar2, width: 300 } }}
```

I made a working example on[ Ractive playground](https://ractive.js.org/playground/index.html?env=docs&hidemenu#N4IgFiBcoE5SAbAhgFwKYGcUgL4BoQBbKARgIDMMoBtUAOyULXgEs6ATNADwDoArKgQDGAezrpx8UXSwACAIIAHRbIC8sgEpIhKFgDc0PbhPYAKYAB06FlOkKLk6SLIAGV2bIA8YEgD4AEmgICCKeAPQ+vu6y0cDAAMSKSDC6SAiyhACeAEYArrZiODjRNgAKyanp0uhcKJAlKJ6KMGi+cQBSAMoA8gByPFgwbADmLOSZpjwAlEXhza3RHote7Pq+GCwAXmjOcQNbaLNhq3pRdB4enie+emm5O7J7twj3R9fLV2sA7izsKGC7YA8H5-MCyAA+4NkAHJOOQkLkECgAOq-f4ANTuaGhbzWsWAYSSKRYaSKVnJ50ewF8WTyBXOWVuMBIZOs52i3hgZ3xvgyOXyKDEj1kPFFjOSACY8LIQf9nABmAAMitkOFVxUp0RceGiQgwGGcLh8wtEIRgzniyoAHCQrUgANyq7W6-UASXYzmhSGU0J1lPYqCQzlMUzUvPMMWsKBsNnFzN2sg222cJEV0ue909KGhqr9MejKDjEoTSYeJAArOmsVmJTn8NEcFM-Y37RTpHJvSp1D8OCIvjxO2pZHQ0F8FMpzNEUMlhmg6jDsiJ2JloVYWxTcABdAhocisDjcfhUHBAA).

Adding code below for reference:
```js
const App = Ractive.extend({
    template: `
  <h1>Hello</h1>
  
  {{#partial mybutton}}
    Partial context:
    <pre>{{JSON.stringify(.)}}</pre>
    
    <div>size: {{.size}}</div>
    <div>value: {{.value}}</div>
    <div>width: {{.width || 'defaultWidthValue'}}</div>
  {{/partial}}

  {{>mybutton myvar1}}

  <hr>

  {{>mybutton { ...myvar2, width: 300 } }}
  
  `,
    css: `h1 { color: #00818a; }`,
    cssId: 'app',
    data: () => ({
        myvar1: {
            size: 10,
            value: 't'
        },
        myvar2: {
            size: 15,
            value: 't2'
        },
    }),
});

const app = window.app = new App({
    target: 'body'
});
```

**Comment by ceremcem** [comment]
@marcalexiei Please show me how you could set the `width` attribute in [this example](https://ractive.js.org/playground/index.html?env=docs&hidemenu#N4IgFiBcoE5SAbAhgFwKYGcUgL4BoQBbKARgIDMMoBtUAOyULXgEs6ATNADwDoArKgQDGAezrpx8AEpIhKFgDc0PUYQAOYtOIzUAOkQCeqjXS0p9AXQAEAXisy5i5dwnsAFMF10rV9OuTokFYABl4+ProoAApIMPJICFai4i6QYVaRADxqMGgAfMDAAFIAygDyAHI8WDBsAOYs5AZuPACUODiZAPQ5+ekR3uGZ7Ip5GCwAXmhBheNTHd0jCnn9VsOjCgkArtNWhZsIOwtdSyuDPuvLAO4s7ChgM8A3d2DHp16RwXgZg+yoSEE3K1bHk9qtcigtjBvJ5zuE5rsAKzfVY+A47IIAZhRcJ8z3uQToWwQCFWOHS5LoOFaHzoySwVgAgmo1LZ7LJ5EoeC4tO5YZE-GoArtQoNMmASHkABJoEkiboSs4DHyFADEali8UShAMACMtigUGIOuksjrjJpxFYETZ9LNJmgTSArOi0LaQPtto7ySAukq9sAepqWAkTXRaSrgHkdfrDWIrDrNjASGGwulxTAzulCqCYwajTCrDxi4nYgAmb74h5WTEABlrVhwjcpypCeHSQgwGCCwQleySIgQIhgQVV9YAHCRx0gANyNr4drsASXYQQA5EgWWv279-oDgTZQR4fpFTyhS8mZtaHUESLXvq71yg142d2fIhey1eEbfkS6vU+ZYvvgFKtDu1IzrSphXEyLIeOkKCxHUaAoOuuoiOwBhrl4EG0rgFgEGg5CsBw3D8FQOBAA). 

Please do not change this part: 
```html
  {{#partial mybutton}}
  	<mycomponent size="{{size}}" value="{{value}}"/> <!-- do not add `width="{{width}}"` attribute here -->
  {{/partial}}
```

**Comment by marcalexiei** [comment]
`mycomponent` declares `width: null` inside its data, so there is no way that Ractive reads ancestors `width` properties and overrides `mycomponent` `width` default value without explicating a value when creating the component instance.

---

To made your example working you have to set `mycomponent`  [`isolated`](https://ractive.js.org/api/#isolated) `flag` to false and remove `width: null` declarartion as seen in this [example](https://ractive.js.org/playground/index.html?env=docs&hidemenu#N4IgFiBcoE5SAbAhgFwKYGcUgL4BoQBbKARgIDMMoBtUAOyULXgEs6ATNADwDoArKgQDGAezrpx8AEpIhKFgDc0PUYQAOYtOIzUAOkQCeqjXS0p9AXQAEAXisy5i5dwnsAFMF10rVlhhHI6OyQVuRICBhoeF4+6OqBaCEABjE+VrooAApIMPLhVqLiLpCpGQA8ajBoAHzAwABSAMoA8gByPFgwbADmLOQGbjwAlDg4ZQD0lTWpPjNWZeyK1RgsAF6JVnUr66MTiwrVcwtLCuEArht1pwgXu+P7h95pxwcA7izsKGAhde+fYHcHl4Mkk8OknuxUEgQm4hrZqps5lUUGcYN5PE80lZthsAKxguY+a4XEIAZgJmKsOFS1LoOCGwLohSwVgAgmo1LZ7LJ5EoeC4tO4MRk4moEslUmUwCRqgAJNAIBAiCbSx6zJ51ADEahyeQQVkIBgARmcUCgxKNSigyobjJpxNi1mgbPotk7LSArMTna7gN6PeM1ZtgJNdSxwpa6IyfHVqoaTWaxAaDKcYCRIzFJWAYI9UrHkwnzeirDxS4bUwAmMF-L5kgAMdapVNp6p8oNSQgwGGS0s2BQCIhgIU1DYAHCRR0gANxU9tPTsYACSwSsAHIkBzV9EIVCYXCbAiPOCMieUOWciQfo71iESHWwd6QquUKuqdvTxlzzAK1ecbf8V65wbM+FavvgNJDNu9JToypivGyHIeKkKA5N0aAoE+RoiOwBirl40GMrgFgEGg5CsBw3D8FQOBAA)

```js
Ractive.components["mycomponent"] = Ractive.extend({
  isolated: false,
  template: `
    mycomponent context:
    <pre>{{JSON.stringify(.)}}</pre>
    
    <div>size: {{size}}</div>
    <div>value: {{value}}</div>
    <div>width: {{width}}</div>
	`, 
  data: () => {
    return {
      size: 5, 
      value: 3, 
    }
  }
});
```

I don't like this approach because when debugging it might be not straightforward understand where `width` value arrives.

**Comment by ceremcem** [comment]
> I don't like this approach because when debugging it might be not straightforward understand where width value arrives.

Exactly. I may simply prefer adding `width="{{width}}"` into the partial declaration in order to be able to easily reason about a problem when I need to.
